data/crac-creation/crac-creator-cim/src/main/java/com/powsybl/openrao/data/craccreation/creator/cim/craccreator/CimCracUtils.java

```diff
@@ -40,4 +52,34 @@ public static Set<FlowCnec> getFlowCnecsFromCrac(MonitoredSeries monitoredSeries
         return mscc.getCreatedCnecIds().stream().map(cnecId -> cracCreationContext.getCrac().getFlowCnec(cnecId)).collect(Collectors.toSet());
     }
 
+    public static String getCurveTypeFromTimeSeries(TimeSeries cimTimeSerie) {
+        String curveType = cimTimeSerie.getCurveType();
+        if (StringUtils.isBlank(curveType)) {
+            curveType = DEFAULT_CURVE_TYPE;
+        }
+        if (!curveType.equals("A03") && !curveType.equals("A01")) {
+            throw new OpenRaoException("CurveType not supported: " + curveType);
+        }
+        return curveType;
+    }
+
+    public static boolean isTimestampInPeriod(Instant timestamp, TimeSeries timeSerie, SeriesPeriod period, int position) {
+        final String curveType = CimCracUtils.getCurveTypeFromTimeSeries(timeSerie);
+        final Duration periodResolution = Duration.parse(period.getResolution().toString());
+        final Instant periodStart = CimCracUtils.parseDateTime(period.getTimeInterval().getStart());
+        final Instant periodEnd = CimCracUtils.parseDateTime(period.getTimeInterval().getEnd());
+
+        Instant pointStart = periodStart.plus(periodResolution.multipliedBy(position - 1L));
+        // default "A03"
+        Instant pointEnd = periodEnd;
```

Comment:
in this case (A03) the ponitEnd is the beginning of the next point (that is the other point with smallest position, such as position > this point's position). if the other point does not exist, then pointEnd=periodEnd. see doc in xsd:
~~~xml
            <CodeDescription>
              <Title>Variable sized Block</Title>
              <Definition>The curve is made of successive Intervals of time (Blocks) of variable duration (size), where the end date and end time of each Block are equal to the start date and start time of the next Interval. For the last Block the end date and end time of the last Interval would be equal to EndDateTime of TimeInterval.</Definition>
            </CodeDescription>
~~~

Reply:
I agree but this isn't the way it is coded in powsybl for glsk management. Should would open a bug ?